client/contract.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/avalanche-rosetta/cache"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+const (
+	contractCacheSize = 1000
+)
+
+// ContractClient is a client for the calling contract information
+type ContractClient struct {
+	ethClient *ethclient.Client
+	cache     *cache.LRU
+}
+
+// NewContractClient returns a new ContractInfo client
+func NewContractClient(endpoint string) (*ContractClient, error) {
+	endpoint = strings.TrimSuffix(endpoint, "/")
+
+	c, err := ethclient.Dial(fmt.Sprintf("%s%s", endpoint, prefixEth))
+	if err != nil {
+		return nil, err
+	}
+
+	cache := &cache.LRU{Size: contractCacheSize}
+
+	return &ContractClient{
+		ethClient: c,
+		cache:     cache,
+	}, nil
+}
+
+// ContractInfo returns the ContractInfo for a specific address
+func (c *ContractClient) ContractInfo(contractAddress common.Address, isErc20 bool) (*ContractInfo, error) {
+	cachedInfo, isCached := c.cache.Get(contractAddress)
+
+	if isCached {
+		return cachedInfo.(*ContractInfo), nil
+	}
+
+	token, err := NewContractInfoToken(contractAddress, c.ethClient)
+	if err != nil {
+		return nil, err
+	}
+	symbol, symbolErr := token.Symbol(nil)
+	decimals, decimalErr := token.Decimals(nil)
+
+	// Any of these indicate a failure to get complete information from contract
+	if symbolErr != nil || decimalErr != nil || symbol == "" || decimals == 0 {
```

Comment:
IMO, if we can't populate the symbol or decimals, we should not index the transfer.

Reply:
We handle the indexing downstream of this, we'll still want to add an entry in the cache so that we can avoid un-necessary rpc calls